thrust/thrust/system/cuda/detail/sort.h

```diff
@@ -109,10 +118,19 @@ THRUST_RUNTIME_FUNCTION cudaError_t doit_step(
   cudaStream_t stream,
   thrust::detail::integral_constant<bool, true> /* sort_items */)
 {
-  using DispatchMergeSortT = cub::DispatchMergeSort<KeysIt, ItemsIt, KeysIt, ItemsIt, Size, CompareOp>;
+  cudaError_t status = cudaSuccess;
+
+  using dispatch32_t = cub::DispatchMergeSort<KeysIt, ItemsIt, KeysIt, ItemsIt, std::uint32_t, CompareOp>;
+  using dispatch64_t = cub::DispatchMergeSort<KeysIt, ItemsIt, KeysIt, ItemsIt, std::uint64_t, CompareOp>;
```

Comment:
No change requested, but some food for thought:

This will instantiate both versions of `DispatchMergeSort` even though we might only ever want to choose one of them in `THRUST_UNSIGNED_INDEX_TYPE_DISPATCH2`

We should design a way to avoid that

Reply:
Agreed! Historically, that's the way it has been: thrust employs a runtime dispatch mechanism, which, at run time chooses the code path depending on the _value_ of the `number of items`, whereas CUB uses a static dispatch mechanism, which decides at compile-time, depending on the _type_ used to pass the `number of items`.

We do have an issue to make this more flexible in thrust ([issue](https://github.com/NVIDIA/cccl/issues/49)):
> We should consider enabling users to configure the offset type at a global scope (e.g., through a preprocessor definition) as well as at a per-algorithm scope. For per-algorithm specification, a likely solution is to allow controlling the offset type via the execution policy, e.g., by adding a offset_type member type definition.